File: documentation-examples-java/src/java/dev/typr/foundations/docs/landing/DuckDbArray.java
package dev.typr.foundations.docs.landing;

import dev.typr.foundations.DuckDbTypes;
import dev.typr.foundations.Fragment;
import dev.typr.foundations.RowCodec;
import dev.typr.foundations.Transactor;
import java.util.List;

@SuppressWarnings("unused")
public class DuckDbArray {
  Transactor tx = null; // placeholder

  // start
  // DuckDB LIST columns are first-class typed values
  List<List<String>> getTagSets() {
    return Fragment.of("SELECT tags FROM posts WHERE published = true")
        .query(RowCodec.of(DuckDbTypes.varchar.list()).all())
        .transactRead(tx);
  }
  // stop
}

File: documentation-examples-scala/src/scala/dev/typr/foundationssc/docs/landing/DuckDbArray.scala
package dev.typr.foundationssc.docs.landing
import dev.typr.foundationssc.*
import dev.typr.foundationssc.Fragment.sql

@SuppressWarnings(Array("unused"))
object DuckDbArray:
  val tx: Transactor = null // placeholder

  // start
  // DuckDB LIST columns are first-class typed values
  def getTagSets(): List[List[String]] =
    sql"SELECT tags FROM posts WHERE published = true"
      .query(RowCodec.of(DuckDbTypes.varchar.list).all())
      .transact(tx)
  // stop
